clang/include/clang/Sema/Sema.h

```diff
@@ -1183,7 +1181,7 @@ class Sema final : public SemaBase {
   std::optional<std::unique_ptr<DarwinSDKInfo>> CachedDarwinSDKInfo;
   bool WarnedDarwinSDKInfoMissing = false;
 
-  bool WarnedStackExhausted = false;
+  SingleWarningStackAwareExecutor StackAwareExecutor;
```

Comment:
Yes, I considered that and it seems somewhat to add some complexity.
Assuming the same code might hit this in both Sema and CodeGen, I don't see a reason the user should see this warning twice.
If Sema and CodeGen would typically be hitting different use cases, it might make sense, but I don't think that's the case.
It's anyways beyond the scope of this change, and we could consider it as a future improvement, though I doubt it should be high priority.

Reply:
Makes sense to keep this as an action for the future. Sharing the logic makes sense, though.

Should we file an issue on GitHub and/or add a FIXME to the code that we might want to share a single instance Sema and CodeGen in the future?

Reply:
Abstractly, we can generate codes without constructing a Sema by loading the AST file by ASTUnit and generate it. So I'd like to make them seperate.

Reply:
I agree that we should still have a warning if we only run CodeGen or only run Sema, and I agree that ideally we should only have a single warning and not potentially two.

Created https://github.com/llvm/llvm-project/issues/112522